src/main.dev.ts

```diff
@@ -188,15 +198,16 @@ app.on('ready', async () => {
     }
   });
 
-  globalShortcut.register(hotKey, () => {
-    if (mainWindow === null) {
-      createRecordingWindow();
-      showNotification('Recording Started', 'Transcibing your voice...');
-    } else {
-      mainWindow.webContents.send('recording:stop', true);
-    }
+  ipcMain.on('hotKey:update', async (_event, data) => {
+    globalShortcut.unregisterAll();
+    globalShortcut.register(data, () => {
+      startRecording();
+    });
+    preferences.updateHotKey(data);
```

Comment:
Since this is an async operation, we might want to `await` this before proceeding to work with `globalShortcut`. Ideally, the error should be caught (async/await supports the try-catch syntax) and forwarded back to the MenuBar instance. If forwarding the error back to the MenuBar is not feasible as due to how the current MenuBar instance is managed, we can mark this segment with a `TODO: Refactor to relay error back to MenuBar.` comment and refactor another time.

e.g.
```ts
ipcMain.on('hotKey:update', async (event, newHotKey) => {
  try {
    await preferences.updateHotKey(hotKey);
    globalShortcut.unregisterAll();
    globalShortcut.register(newHotKey, startRecording);
  } catch (err) {
    // Alert MenuBar instance of error via IPC.
    event.reply('hotKey:fail', err);
  }
});

```